### PR TITLE
Basic implementation of a graph view

### PR DESCRIPTION
**Pull request by gaspardthrl**
Implementation of a force-directed graph visualization with some gestures handlers like dragging the graph, dragging a specific node, and zooming / dezooming. Graph.tsx represents a basic graph structure G = (V, E) ; ForceDirectedGraph.tsx contains the logic for creating and displaying a force-directed graph ; fruchtermanReingold.tsx contains the logic for the Fruchterman-Reingold Algorithm used to represent force-directed graphs.

Note (will be dealt with with in next pull request(s)):

Several things still need to be implemented among which:
- Rotation Gesture
- Graph initialization loop view
- Impact of dragging a node on others

Important things to work on:
- Documentation
- Code readability
- Modularization

### CODE REVIEW

File: graph/ForceDirectedGraph/ForceDirectedGraph.tsx [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+    );
+    setSizes(initialSizes);
+    setLoad(true);
+    return () => {
+      setLinks([]);
+      setNodes([]);
+      setSizes(new Map());
+      setLoad(false);
+    };
+  }, [graph, constrainedNodeId, screenWidth, screenHeight]);
```
Comment (Aidasvenc) [inline graph/ForceDirectedGraph/ForceDirectedGraph.tsx:75]: screenWidth and screenHeight normally should not change. I think you can safely remove them from the sensitivity list

File: graph/ForceDirectedGraph/ForceDirectedGraph.tsx [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+  };
+
+  // If the graph is not loaded, display a loading message
+  if (!load) {
+    return (
+      <View style={styles.container}>
+        <Text>Loading...</Text>
+      </View>
+    );
+  }
```
Comment (Aidasvenc) [inline graph/ForceDirectedGraph/ForceDirectedGraph.tsx:146]: For this, we should use an `ActivityIndicator` component. [ActivityIndicator](https://reactnative.dev/docs/activityindicator)

File: graph/ForceDirectedGraph/ForceDirectedGraph.tsx [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+    >
+      {PAN_GESTURE_HANDLER}
+    </PinchGestureHandler>
+  );
+  return (
+    <GestureHandlerRootView style={styles.container}>
+      {PINCH_GESTURE_HANDLER}
+    </GestureHandlerRootView>
+  );
+};
```
Comment (Aidasvenc) [inline graph/ForceDirectedGraph/ForceDirectedGraph.tsx:239]: I like your personal choice of modularizing everything, but if there are many parent components, it becomes a bit hard to visually see the hierarchy of components. I suggest returning plain pinch and gesture handlers together with the graph canvas in the final return statement. Only Circles and Lines should be defined externally. However, that's just my opinion, either way works.

**Review by Aidasvenc — COMMENTED** [review]
Very nice coding style from your part. I really enjoy that you created class components to deal with the graph. Im sure it is going to turn out really well!

**Review by Fyuryy — COMMENTED** [review]
LGTM!

**Pull request closed without merging**